suite-common/wallet-core/src/send/sendFormCardanoThunks.ts

```diff
@@ -13,37 +11,48 @@ import {
 } from '@suite-common/wallet-utils';
 import { notificationsActions } from '@suite-common/toast-notifications';
 import {
-    Account,
     PrecomposedLevelsCardano,
     PrecomposedTransactionCardano,
 } from '@suite-common/wallet-types';
 import { createThunk } from '@suite-common/redux-utils';
 
-import { selectDevice } from '../device/deviceReducer';
-import { ComposeTransactionThunkArguments } from './sendFormTypes';
+import {
+    ComposeTransactionThunkArguments,
+    ComposeFeeLevelsError,
+    SignTransactionError,
+    SignTransactionThunkArguments,
+} from './sendFormTypes';
 import { SEND_MODULE_PREFIX } from './sendFormConstants';
 
-export const composeCardanoSendFormTransactionThunk = createThunk(
-    `${SEND_MODULE_PREFIX}/composeCardanoSendFormTransactionThunk`,
-    async ({ formValues, formState }: ComposeTransactionThunkArguments, { dispatch }) => {
-        const { account, feeInfo } = formState;
+export const composeCardanoTransactionFeeLevelsThunk = createThunk<
+    PrecomposedLevelsCardano,
+    ComposeTransactionThunkArguments,
+    { rejectValue: ComposeFeeLevelsError }
+>(
+    `${SEND_MODULE_PREFIX}/composeBitcoinTransactionFeeLevelsThunk`,
+    async ({ formState, composeContext }, { dispatch, rejectWithValue }) => {
+        const { account, feeInfo } = composeContext;
         const changeAddress = getUnusedChangeAddress(account);
-        if (!changeAddress || !account.utxo || !account.addresses) return;
+        if (!changeAddress || !account.utxo || !account.addresses)
```

Comment:
are we sure everything works on desktop now with rejecting instead of plain return?

Reply:
yes, it was tested.